### DISCUSSION TITLE
How to style the options in a Select widget?

### DISCUSSION THREAD

**Discussion by UlrichGoebel** [Textualize/textual#4329] [General]
As I asked in https://github.com/Textualize/textual/discussions/4304 for Buttons I want to build single line Select widgets. It seems that the options themselves has `border`, `padding` or even other properties, but I don't find a way to configure them.

My little code snipped inside a `compose`:

```python
choices = (
    ('Choice One', 'c1'),
    ('Choice Two', 'c2'),
    ('Choice Three', 'c3'),
)
yield Select(choices, allow_blank=True, prompt='make your choice')
```

That makes a three-line widget, even if I set the `border` to none, and so on. Even an additional border is shown if I set `border: red` for the Select widget. That's why I suppose that the options themselve have `border` or `margin` or something else.

A detail could be: how to style the `promt` in case of no selection?

Any help is very welcome.

**Comment by TomJGooding** [comment]
Does this answer in a similar discussion help? https://github.com/Textualize/textual/discussions/4061#discussioncomment-8218119

**Reply by UlrichGoebel** [reply]
That solves my question. Thank's a lot!

**Comment by UlrichGoebel** [comment]
Thank's to GomjGooding, the discussion #4061 solves the question.

As a beginner I am proud (😉) to found how to restrict these settings to a specific Select. That could be done by CSS using [child combinators](https://textual.textualize.io/guide/CSS/#child-combinator).

Let's assume we have a Select with `id='onelineselect'` where we want to eliminate the border to get a one-line-Select. Then we could make a CSS file with the snippet:

```CSS
#onelineselect > SelectCurrent, #onelineselect > SelectCurrent:focus {
    border: none;
    }
```

Then as the python code snippet we could have:

```python
choices = (('A', 'a'), ('B', 'b'))
yield Select(choices, id='onelineselect', allow_blank=True, prompt='make your Choice')
```

Of course the `allow_blank` and `promt=...` are optional and are intended to demonstrate that even the prompt for the blank selection is covered to be one-line.
